### ISSUE TITLE
[Question] Clarification on total training time with 5-fold cross-validation

### ISSUE THREAD

**Issue by dvpalomino** [MIC-DKFZ/nnDetection#314]
### :question: Question
Hi, thank you for your great work on nnDetection!

I have a question regarding the estimated training time. According to the documentation, training takes approximately 2 hours per epoch with 3D mixed precision on an RTX 2080Ti, for a total of ~2 days for standard training (assuming 24–30 epochs).

However, if we run 5-fold cross-validation, would the total training time scale linearly? For example:

60 epochs × 5 folds × 2 hours/epoch = 600 hours (~25 days).

Is this estimation accurate, or are there mechanisms in place (e.g., early stopping, fold parallelization) that would reduce the overall training time?

Thanks in advance for the clarification!

**Comment by dvpalomino** [comment]
Thanks for your reply. I was able to generate the training curves for the first fold (60 epochs), and from what I can see, I don't think overfitting is occurring — the training and validation losses remain in a similar range. However, I do observe that the detection metrics stop improving and tend to stabilize at a certain point during training.
Based on the attached plots, would it make sense to reduce the number of epochs (e.g., stop around epoch 30) to save training time without affecting performance?

Also, I’ve noticed some other metrics being logged, such as seg_dice, val_seg_dice, seg_ce, val_seg_ce, train_cls, val_cls, train_reg, and val_reg. I’m not entirely sure what each of these refers to. For instance, in my case, the seg_dice value decreases from around 0.5 to 0.2 over the 60 epochs — is that expected behavior?

![Image](https://github.com/user-attachments/assets/6f4a6603-ce83-4852-aaf8-cd32305d9c4e)

![Image](https://github.com/user-attachments/assets/552788d7-b333-49f9-831f-b013fe2e7d09)

![Image](https://github.com/user-attachments/assets/8f99f7cd-227f-4ccc-8ee8-c459414a6855)

![Image](https://github.com/user-attachments/assets/58a1f1f8-8295-4b55-9dd9-e2af1c32a85a)

**Comment by andres2631996** [comment]
Hello, so you could do the following, provided that your pretrained architecture would match in size with the architecture you want to train:

In [https://github.com/MIC-DKFZ/nnDetection/blob/98eb97f15d9f9d107fac11d27c153e18932ab635/scripts/train.py#L270](url) insert before the call to pl.Trainer:

`module = module.load_from_checkpoint("path/to/checkpoint.ckpt")`

You would also need to adjust the learning rate and number of epochs in [https://github.com/MIC-DKFZ/nnDetection/blob/main/nndet/conf/train/v001.yaml](url) to accommodate your fine tuning schedule.

Additionally, I think the thread here would help you: #172, where it is also discussed how to adjust the target spacing of the pretrained model to the target spacing of your main model.

Unfortunately, there is not an automated way to do this more quickly.
